text/0004-optional-frontend.md

```diff
@@ -0,0 +1,48 @@
+- Title: Optional Meilisearch Front-end
+- Start Date: 2020/11/16
+- specification PR: #8
+- Meilisearch Issue: #411
+
+# Optional Meilisearch Front-end
+
+## First section: Feature Description and Interaction
+
+### Summary
+
+For testing purposes, MeiliSearch is shipped with a frontend that is served at the server's root. This provides the user with a basic GUI to perform searches on his documents. While this is useful in a development environment, this is not necessarily desirable in a production environment. This specification proposes to remove this frontend in a development environment.
+
+### Motivation
+
+This frontend was developed for testing purposes, and it should not be present in a production environment.
+
+### Prior Art and R&D
+
+N.A
+
+### Explanation
+
+When the option `--env` or `MEILI_ENV` is set to `production`, the frontend is disabled.
+
```

Comment:
```suggestion
When the option `--env` or `MEILI_ENV` is set to `production`, the frontend is disabled unless the option `--enable-gui` or `MEILI_ENABLE_GUI` is set to `true`.
```

Reply:
Can't we wait until someone complains until we introduce a new option? I like @LegendreM because it doesn't introduce any change to the interface, and t probably fits in 99,9% of the usecases, wdyt @eskombro, @curquiza ?

Reply:
I'm not against Sam's suggestion, but on the other hand, I also think if a user wants the interface in production, this is not production, this is for dev purposes.
So, I'm more on Marin's side: wait for complaints and specific use-cases before adding this option. But this is just my opinion.

Reply:
No matter what, adding an option does not change anything to what is done here, so if we all agree on this, we can merge this spec. If adding an option is needed, it can be done in another RFC, without conflicts with this one.

Reply:
> Can't we wait until someone complains

Ok, I am mainly saying this thinking on Sandbox users first (always transposing that to any other possible automation tools, like DO Image, Saas, and this kind of things) where the USER is not the OWNER. 

- I can set a production environment to give production-ready instances to someone else, it doesn't suit me to not have the dashboard as it is the only visible+interactive part of the running MeiliSerch instance in the browser (doesn't require a terminal or any other normal tool).
- I do a demo, I want to set it In production, I can still use that web GUI. 
- I do a tutorial, I want it activated to have a GUI.

Why are you taking this from me? 😢 

[ok @MarinPostma , someone has complained now haha ;) ]

Reply:
I am only asking for this if there is no implementation complexity

Reply:
What's the difference between prod and dev env again?

Reply:
Actually checking the [docs](https://docs.meilisearch.com/guides/advanced_guides/configuration.html#environment) it is not quite much of a difference. But I don't know what will be built on top of that.

Reply:
@eskombro Is it ok for you to switch to a dev env, until we figure out what we want to do with this settings?

Reply:
Yes, let's not block this, go ahead :)